public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/PyroscopeAppIntegration.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useEffect, useMemo, useState} from 'react';
+
+import { QueryEditorProps, TimeRange, DateTime } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import {LinkButton} from '@grafana/ui'
+
+import { PyroscopeDataSource } from '../datasource';
+import { PyroscopeDataSourceOptions, Query } from '../types';
+
+const PYROSCOPE_APP_ID = 'grafana-pyroscope-app'
+
+/* Global promise to fetch the pyroscope app settings */
+let pyroscopeAppSettings: Promise<PyroscopeAppSettings> | null= null;
+/* Global promises to fetch pyroscope datasource settings by uid as encountered */
+const pyroscopeDatasourceSettingsByUid: Record<string, Promise<PyroscopeDatasourceSettings>> = {};
+
+export type Props = QueryEditorProps<PyroscopeDataSource, Query, PyroscopeDataSourceOptions>;
+
+/** A subset of the app settings that are relevant for this integration */
+type PyroscopeAppSettings = {
+  enabled: boolean;
+  jsonData: {
+    backendUrl: string;
+    basicAuthUser: string;
+  }
+}
+
+/** A subset of the datasource settings that are relevant for this integration */
+type PyroscopeDatasourceSettings = {
+  url: string;
+  basicAuthUser: string | number; // This might be configured as a number, so we will want to ensure compare for string-equivalence
```

Comment:
Why would this be a number?

Reply:
This was to accommodate a potential configuration error.
It SHOULD be a string, but sometimes when provisioning people might use a plain number to represent the instance in the yaml vs. a quoted number.
But come to think of it, by the time the backend code provides this to the frontend, it will have been converted to a string.
I will remove this accommodation, it is not needed.